doc/flow_counters/routes_flow_counters.md

```diff
@@ -0,0 +1,475 @@
+
+For binding route entry to a counter:
+
+1. Request Flow Counter Handler to create a generic counter
+2. Bind the counter to the route
+3. Request FlexCounterManager to set the counter Id list to FLEX_COUNTER_TABLE so that syncd knows how to query the stats
+4. Save a new entry {<prefix_str>:<counter_id>} and {<prefix_str>:<route_pattern>} to COUNTERS DB so that CLI knows how to display the statistic.
+
+For unbinding route entry from a counter:
+
+1. Request FlexCounterManager to remove the counter Id list from FLEX_COUNTER_TABLE
+2. Unbind the route and counter
+3. Request Flow Counter Handler to remove the generic counter
+4. Remove the entry from COUNTERS DB
+
+#### Counters DB Enhancements
+
+A new table COUNTERS_ROUTE_NAME_MAP shall be added to COUNTERS DB.
+
+	; Defines information for COUNTERS_ROUTE_NAME_MAP
+	key                     = COUNTERS_ROUTE_NAME_MAP     ; route string to generic counter OID mapping
+	; field                 = value
+    ...
+	<str_of_route>          = STRING                      ; field name is string of route, value is counter OID
+
+Example:
+
+```
+COUNTERS_ROUTE_NAME_MAP: {
+"1.1.1.0/24:Vrf_default":"oid:0x1500000000034e"
+"1.1.7.7/32:Vrf_default":"oid:0x1500000000035e"
+}
+```
+
+A new table COUNTERS_ROUTE_TO_PATTERN_MAP shall be added to COUNTERS DB.
+
+	; Defines information for COUNTERS_ROUTE_TO_PATTERN_MAP
+	key                     = COUNTERS_ROUTE_TO_PATTERN_MAP     ; route string to route pattern mapping
+	; field                 = value
+    ...
+	<str_of_route>          = STRING                            ; field name is string of route, value is route pattern
+
+Example:
+
+```
+COUNTERS_ROUTE_TO_PATTERN_MAP: {
+"1.1.1.0/24:Vrf_default":"1.1.0.0/16:Vrf_default"
+"1.1.7.7/32:Vrf_default":"1.1.0.0/16:Vrf_default"
+}
+```
+
+### SAI API
+
+No new SAI API is required for this feature.
+
+SAI APIs shall be used in this feature:
+
+ |API                 |Function                           |
+ |:------------------:|-----------------------------------|
+ |sai_counter_api_t   |create_counter                     |
+ |                    |remove_counter                     |
+ |                    |get_counter_stats_ext              |
+ |sai_route_api_t     |set_route_entries_attribute        |
+
+SAI attributes shall be used in this feature:
+
+ |Struct                |Attribute                          |
+ |:--------------------:|-----------------------------------|
+ |sai_counter_stat_t    | SAI_COUNTER_STAT_PACKETS          |
+ |                      | SAI_COUNTER_STAT_BYTES            |
+ |sai_route_entry_attr_t| SAI_ROUTE_ENTRY_ATTR_COUNTER_ID   |
+
+### Configuration and management
+
+#### CLI/YANG model Enhancements
+
+##### CLI
+
+Enable/disable configuration:
+
+```
+counterpoll flowcnt-route  <enable | disable>
+Example:
+admin@sonic:~$ counterpoll flowcnt-route enable
+```
+
+Polling interval configuration:
+
+```
+counterpoll flowcnt-route interval <time_in_msec> // default - 1000ms
```

Comment:
Hope we'll generate the Click commands using auto generation tool (https://github.com/Azure/SONiC/pull/780)

Reply:
as long as there is a yang module for it you ca use the auto generation tool. but it is not done via the build system.
it will be invoked manually and added like it was coded :-). if this is what you mean that we are aligned and comment can be resolved.